docs/docs/metrics.md

```diff
@@ -4,128 +4,256 @@ title: Metrics
 
-## Instrumented method calls
+## Prometheus reporter
 
-These metrics are created automatically by instrumenting certain classes in our code base.
+The Prometheus reporter can be enabled with `--metrics_prometheus`
+command-line flag. When enabled, metrics in the Prometheus format are
+available at `/metrics/prometheus`.
 
-You can disable these instrumented metrics with `--disable_metrics`. This flag will only disable this code instrumentation, not all metrics.
+Dots and dashes in metric names are replaced with underscores.
 
-These timers can be very valuable in diagnosing problems, but they require detailed knowledge of the inner workings of Marathon. They can also degrade performance noticeably.
+## StatsD reporter
 
-Since these metric names directly correspond to class and method names in our code base,
-expect the names of these metrics to change if the affected code changes.
+The StatsD reporter can be enabled with `--metrics_statsd` command-line
+flag. It sends metrics over UDP to the host and port specified with
+`--metrics_statsd_host` and `--metrics_statsd_port` respectively.
 
-## Potential pitfalls
+## DataDog reporter
 
-### Derived metrics (mean, p99, ...)
+The DataDog reporter can be enabled with `--metrics_datadog`
+command-line flag. It sends metrics over UDP to the host and port
+specified with `--metrics_datadog_host` and `--metrics_datadog_port`
+respectively.
 
-Our metrics library calculates derived metrics like "mean" and "p99" using a sliding average window algorithm. This means that every time you fetch the `/metrics` endpoint you are looking at the average of the last N seconds. By default the length of the window is 30 seconds, but this can be configured with the `--metrics_averaging_window` flag.
+Marathon can send metrics to a DataDog agent over UDP, or directly to
+the DataDog cloud over HTTP. It is specified using
+`--metrics_datadog_protocol`. Its possible values are `udp` (default)
+and `api`.
 
-For getting the most accurate results it is recommended to configure your polling interval to the size of the sliding average window.
+Dashes in metric names are replaced with underscores.
 
-### Statsd and derived statistics
+## Important metrics
 
-Statsd typically creates derived statistics (mean, p99) from mean values Marathon reports. Our metrics package also reports derived statistics. To avoid accidentally aggregating statistics multiple times, be sure you know where you are reporting and computing mean values.
+* `marathon.apps.active.gauge` — the number of active apps.
+* `marathon.deployments.active.gauge` — the number of active
+  deployments.
+* `marathon.deployments.counter` — the count of deployments received
+  since the current Marathon instance became a leader.
+* `marathon.deployments.dismissed.counter` — the count of deployments
+  dismissed since the current Marathon instance became a leader.
+* `marathon.groups.active.gauge` — the number of active groups.
```

Comment:
Our naming conventions. `count` alludes a counter. And it is not just `groups`, because this way we name counters. In the future there might be such one, and it will mean the total number of groups created so far by the current Marathon instance.